### ISSUE TITLE
Assume role arn feature in TupleS3StoreBackend

### ISSUE THREAD

**Issue by Reactor11** [great-expectations/great_expectations#7546]
I'm hosting my GX components(Validations, Data Docs etc) in s3 using TupleS3StoreBackend in different aws account (Say PROD), Now I want to use the same GX context in different aws account (Say PCI), Since using IAM role is the recommended way instead of using IAM user.

We should be able to give which role to assume while writing in [boto_options](https://docs.greatexpectations.io/docs/guides/setup/configuring_metadata_stores/how_to_configure_an_expectation_store_in_amazon_s3/#3-update-your-configuration-file-to-include-a-new-store-for-expectations-on-s3) and it should automatically assume the role and write to s3.

Alternatives that I have consider is, I've a separate flow that will assume role and set the credentials in env, post which i'll load GX context

Example config right now - 

```
checkpoint_store:
    class_name: CheckpointStore
    store_backend:
      class_name: TupleS3StoreBackend
      bucket: bucket
      prefix: checkpoint/
      boto3_options:
        aws_access_key_id: ${DQ_SESSION_ID}
        aws_secret_access_key: ${DQ_SESSION_KEY}
        aws_session_token: ${DQ_SESSION_TOKEN}
```

Example config expected:
```
checkpoint_store:
    class_name: CheckpointStore
    store_backend:
      class_name: TupleS3StoreBackend
      bucket: bucket
      prefix: checkpoint/
      boto3_options:
        aws_assume_role_arn: ${role-arn}
```

**Comment by Reactor11** [comment]
PR - https://github.com/great-expectations/great_expectations/pull/7547

**Comment by rdodev** [comment]
Hey @Reactor11 This is great. I'll ask engineer to 👀

**Comment by rdodev** [comment]
@Reactor11 engineering has reviewed the PR and left some comments for you. Not sure if you've gotten the chance to check on those.

**Comment by Reactor11** [comment]
hey @rdodev I've made the suggested changes.

**Comment by austiezr** [comment]
Closing as resolved by #7547

**Comment by Reactor11** [comment]
hey @rdodev / @austiezr - I've added documentation for the same [here](https://github.com/great-expectations/great_expectations/pull/7634)
